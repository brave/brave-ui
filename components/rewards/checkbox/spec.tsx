/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Checkbox from './index'

describe('List tests', () => {
  const baseComponent = (props?: object) => <Checkbox
    value={{'yt': true, 'tw': false, 'inst': false}}
    id={'checkbox'}
    multiple={false}
    title={'Enable ability to give tips on ‘Like’ posts'}
  >
    <div data-key='yt'>YouTube</div>
    <div data-key='tw'>Twitter</div>
    <div data-key='inst'>Instagram</div>
  </Checkbox>

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#checkbox').length
      expect(assertion).toBe(1)
    })
  })
})
