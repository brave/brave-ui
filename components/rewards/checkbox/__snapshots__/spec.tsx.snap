// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List tests basic tests matches the snapshot 1`] = `
.c0 {
  width: 100%;
  margin-bottom: 12px;
}

.c1 {
  width: 100%;
  font-family: Poppins;
  line-height: normal;
  font-size: 14px;
  font-weight: 500;
  color: #686978;
  margin-bottom: 6px;
}

.c2 {
  font-family: Poppins;
  font-size: 14px;
  line-height: 2.79;
  color: #686978;
}

.c3 {
  width: 18px;
  height: 18px;
  border-radius: 2px;
  border: solid 1px #d1d1db;
  display: inline-block;
  margin-right: 11px;
  position: relative;
  top: 3px;
  border-color: #a1a8f2;
  background: url(test-file-stub) no-repeat 3px 5px;
}

.c4 {
  width: 18px;
  height: 18px;
  border-radius: 2px;
  border: solid 1px #d1d1db;
  display: inline-block;
  margin-right: 11px;
  position: relative;
  top: 3px;
}

<div
  className="c0"
  id="checkbox"
>
  <div
    className="c1"
  >
    Enable ability to give tips on ‘Like’ posts
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <span
      className="c3"
      selected={true}
    />
     
    YouTube
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <span
      className="c4"
      selected={false}
    />
     
    Twitter
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <span
      className="c4"
      selected={false}
    />
     
    Instagram
  </div>
</div>
`;
